### DISCUSSION TITLE
Tip:  Faster debugging of themes

### DISCUSSION THREAD

**Discussion by markdav-is** [oqtane/oqtane.framework#718] [General]
# Debuging Themes

debugging themes using multiple solutions is a slow and painful process.  This is a shortcut that should save you lots of time when developing themes.

### Prerequisites

- [ ]  Visual Studio
- [ ]  Oqtane.Framework repo and the Oqtane solution
- [ ]  Oqtane.Theme.[your theme] repo and project

It's a great idea to have these two repos as siblings in the same root folder.

### Setup

1. Open the Oqtane solution and add a new existing project 

![image](https://user-images.githubusercontent.com/311063/91644923-9ee09880-e9f5-11ea-95b2-f2d1316c83d2.png)

2.  select the csproj file for your theme

![image](https://user-images.githubusercontent.com/311063/91644936-b4ee5900-e9f5-11ea-9c70-9b19e5dea03d.png)

Debugging

1. make a chance to your Theme 

![image](https://user-images.githubusercontent.com/311063/91644943-c3d50b80-e9f5-11ea-8e54-4d76b0f7bae5.png)

2. build or rebuild the SOLUTION

![image](https://user-images.githubusercontent.com/311063/91644950-cfc0cd80-e9f5-11ea-939b-a1c8de467ef3.png)

3. run the Oqtane.Server project like you normally would
![image](https://user-images.githubusercontent.com/311063/91644956-dd765300-e9f5-11ea-8ea9-edf6bf28c9fa.png)

4. see your change reflected.

![image](https://user-images.githubusercontent.com/311063/91644967-f1ba5000-e9f5-11ea-8b5d-4ccac9ec087e.png)

you will need to rebuild the solution every time to see you changes.

Packaging themes and deploying them will still be done the reqular way, but this should speed up the debug cycle.

**Comment by PizzaConsole** [comment]
I have been doing this for my external modules

**Reply by markdav-is** [reply]
great.  Im loing at this today and hope I can use it for debugging as well.
